js/binance.js

```diff
@@ -271,19 +295,98 @@ module.exports = class binance extends Exchange {
+        //                     {"filterType":"LOT_SIZE","minQty":"0.00100000","maxQty":"100000.00000000","stepSize":"0.00100000"},
+        //                     {"filterType":"MIN_NOTIONAL","minNotional":"0.00010000","applyToMarket":true,"avgPriceMins":5},
+        //                     {"filterType":"ICEBERG_PARTS","limit":10},
+        //                     {"filterType":"MARKET_LOT_SIZE","minQty":"0.00000000","maxQty":"63100.00000000","stepSize":"0.00000000"},
+        //                     {"filterType":"MAX_NUM_ALGO_ORDERS","maxNumAlgoOrders":5}
+        //                 ]
+        //             },
+        //         ],
+        //     }
+        //
+        // futures (fapi)
+        //
+        //     {
+        //         "timezone":"UTC",
+        //         "serverTime":1575417244353,
+        //         "rateLimits":[
+        //             {"rateLimitType":"REQUEST_WEIGHT","interval":"MINUTE","intervalNum":1,"limit":1200},
+        //             {"rateLimitType":"ORDERS","interval":"MINUTE","intervalNum":1,"limit":1200}
+        //         ],
+        //         "exchangeFilters":[],
+        //         "symbols":[
+        //             {
+        //                 "symbol":"BTCUSDT",
+        //                 "status":"TRADING",
+        //                 "maintMarginPercent":"2.5000",
+        //                 "requiredMarginPercent":"5.0000",
+        //                 "baseAsset":"BTC",
+        //                 "quoteAsset":"USDT",
+        //                 "pricePrecision":2,
+        //                 "quantityPrecision":3,
+        //                 "baseAssetPrecision":8,
+        //                 "quotePrecision":8,
+        //                 "filters":[
+        //                     {"minPrice":"0.01","maxPrice":"100000","filterType":"PRICE_FILTER","tickSize":"0.01"},
+        //                     {"stepSize":"0.001","filterType":"LOT_SIZE","maxQty":"1000","minQty":"0.001"},
+        //                     {"stepSize":"0.001","filterType":"MARKET_LOT_SIZE","maxQty":"1000","minQty":"0.001"},
+        //                     {"limit":200,"filterType":"MAX_NUM_ORDERS"},
+        //                     {"multiplierDown":"0.8500","multiplierUp":"1.1500","multiplierDecimal":"4","filterType":"PERCENT_PRICE"}
+        //                 ],
+        //                 "orderTypes":["LIMIT","MARKET","STOP"],
+        //                 "timeInForce":["GTC","IOC","FOK","GTX"]
+        //             }
+        //         ]
+        //     }
+        //
         if (this.options['adjustForTimeDifference']) {
             await this.loadTimeDifference ();
         }
         const markets = this.safeValue (response, 'symbols');
         const result = [];
         for (let i = 0; i < markets.length; i++) {
             const market = markets[i];
+            const future = ('maintMarginPercent' in market);
+            const spot = !future;
+            const marketType = spot ? 'spot' : 'future';
```

Comment:
@lyn716 it's a perpetual future, right? https://www.binance.vision/economics/what-are-perpetual-futures-contracts

Reply:
@lyn716 as far as i understand, perpetual futures contracts (pnl fixed upon settling) are different from swap trades (pnl fixed regularly). In other words, despite that OKEX perpetual swaps have some commonalities with Binance perpetual futures, they're not the same type. Binance perpetual futures are closer to OKEX futures without a closing date.

Reply:
In Binance, there's only perpetual futures.

In Okex, there's `futures` that settles at some fixed duration and there's `perpetual swap` that behaves more like Binance's perpetual futures.

Reply:
@StanleyDing 

> In Binance, there's only perpetual futures.

Yep, exactly.

> In Okex, there's futures that settles at some fixed duration and there's perpetual swap 

Right.

> and there's perpetual swap that behaves more like Binance's perpetual futures.

The last piece of this sentence sounds a bit wrong to me. Binance perpetual futures are not settled every 12 hours like OKEX perpetual swaps are. Therefore Binance perpetual futures behave **not like OKEX swaps** but **more like OKEX futures without a closing date**. 

Let me know if I'm wrong.

Reply:
@StanleyDing @lyn716 https://medium.com/@OKEx/perpetual-swap-okex-vs-other-exchanges-fbb0533f4141

Reply:
@StanleyDing @lyn716 if you agree to the above – I am going to merge it within an hour or so.

Reply:
I'm saying OKEX's swap is more like Binance in a sense that OKEX swap does not close position on settlement day. However, OKEX does settle every day at 4pm (UTC+8) which moves unrealized pnl to realized pnl. In Binance or Bitmex, pnl is only calculated when you close a position.

Reply:
@StanleyDing in one sense – yes, it's more like OKEX swap, because it does not have a closing date. However, because OKEX does settle it – everything else is more not like a swap. The real question here is what should we put into the `type` of a Binance futures market – `future` or `swap` ?

Reply:
I'm not from a financial background so I am really not sure about the naming. 

In Binance's case, I think I would prefer using `futures` to describe its perpetual futures. In other exchange however, `futures` might cause confusion between futures that closes at specific data and futures that doesn't.

Reply:
By the way, I personally don't think okex swap is that different from perpetual futures in binance or bitmex. They have somewhat similar funding mechanism, liquidation engine and it never closes position. The only difference I think is that OKEX settles at 4 p.m. (UTC+8), but the position is still unaffected.

Reply:
@StanleyDing is that a strong enough reason to put `swap` instead of `future` into binance's markets' type, though?

Reply:
Nope. I would 100% go for `futures` in Binance. 😂

Reply:
If only discuss words' meaning, what I understood is: `Perpetual Futures` == `Swap`. And base on this, I thought `swap` is one special case of `futures`. 

I also agree to set `type` value as `futures` on this. Because BinanceFuture themselves use this word. It depends on ccxt if need unified semantics.

Reply:
@StanleyDing @lyn716 thx, merging it in a few sec and will unify the futures' type word throughout the rest of the exchanges during this week.